src/cli/test/test_with_analysis.py

```diff
@@ -0,0 +1,87 @@
+    suffix = "labs" if is_labs else "classic"
+    osw_path = base_osw_path.parent / f"with_analysis_{suffix}.osw"
+    runDir = base_osw_path.parent / f"run_{suffix}"
+    osw["run_directory"] = str(runDir)
+    runDir.mkdir(exist_ok=True)
+    with open(osw_path, "w") as f:
+        json.dump(osw, fp=f, indent=2, sort_keys=True)
+
+    # Fake having an in.idf or it won't run in the "classic" subcommand, doing it for labs too so that it's less
+    # confusing
+    # if not is_labs:
+    with open(runDir / "in.idf", "w") as f:
+        f.write("Building,;")
+
+    command = [str(osclipath)]
+    if not is_labs:
+        command.append("classic")
+    command += ["run", "--postprocess_only", "-w", str(osw_path)]
+    lines = subprocess.check_output(command, encoding="utf-8").splitlines()
+
+    assert runDir.exists()
+    measure_attributes_path = runDir / "measure_attributes.json"
+    assert measure_attributes_path.is_file()
+    results_path = runDir / "results.json"
+    assert results_path.is_file()
+    objectives_path = runDir / "objectives.json"
+    assert objectives_path.is_file()
+
+    measure_attributes = json.loads(measure_attributes_path.read_text())
+    assert measure_attributes == {
+        "FakeReport": {"applicable": True, "net_site_energy": 167.1, "something_with_invalid_chars": 1}
+    }
+
+    results = json.loads(results_path.read_text())
+    assert results == {"FakeReport": {"applicable": True, "net_site_energy": 167.1, "something_with_invalid_chars": 1}}
+
+    objectives = json.loads(objectives_path.read_text())
+    assert objectives == {
+        "objective_function_1": 167.1,
+        "objective_function_3": 1.7976931348623157e308,
+        "objective_function_group_1": 1.0,
+        "objective_function_group_3": None,
+        "objective_function_target_1": 0.0,
+        "objective_function_target_3": None,
+        "scaling_factor_1": 1.0,
+        "scaling_factor_3": None,
+    }
+
+    expected_files_in_run_dir = {
+        "data_point.zip",
+        "finished.job",
+        "in.idf",
+        "measure_attributes.json",
+        "objectives.json",
+        "results.json",
+        "run.log",
+        "started.job",
+        # TODO: see below
+        "data_point_out.json",
+    }
+    # TODO: I'm letting this test fail so it's obvious this needs to be addressed
+    if True:  # not is_labs:
+        # We get the SAME exact info in measure_attributes.json, results.json and data_point_out.json...
+        # measure_attributes.json is flushed after each apply measure Step (ModelMeasures, EnergyPlusMeasures,
+        # ReportingMeasures), then at the end of ReportingMeasures it's done once again and results.json is spat out too
+        # Do we really need the data_point_out.json in addition to this?
+        # Seems like we could just run the output of results.json/data_point_out.json at the end of the workflow run
+        # instead
+        expected_files_in_run_dir.add("data_point_out.json")
```

Comment:
@kbenne  @brianlball Here is the TODO I want to adress (with a voluntarily failing tests so it doesn't slip by).

Do we really need all three of these json files? if so, when?